### PR TITLE
Add roomDestroyed to MUC UserStatusListener

### PR DESCRIPTION
**Pull request by meisterfuu**
I could not test it yet, but it is pretty simple and looks ok to me.

### CODE REVIEW

**Comment by meisterfuu** [comment]
error: method roomDestroyed in interface UserStatusListener cannot be applied to given types;
                listener.roomDestroyed(mucUser.getDestroy().getJid(), mucUser.getDestroy().getReason());
                        ^
  required: Jid,String
  found: String,String

Fixing this now

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/DefaultUserStatusListener.java [hunk 1]
```diff
@@ -65,5 +65,8 @@ public void adminGranted() {
 
     public void adminRevoked() {
     }
+    
+    public void roomDestroyed(Jid alternateJID, String reason) {
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/DefaultUserStatusListener.java]: Why is the parmater name `alternateJID`?

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java [hunk 2]
```diff
@@ -2181,6 +2182,23 @@ private void checkPresenceCode(
                 listener.nicknameChanged(from, mucUser.getItem().getNick());
             }
         }
+        //The room has been destroyed
+        if (mucUser.getDestroy() != null) {
+            Jid alternateJid = null;
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java]: Why not simply

    alternateJid = mucUser.getDestroy().getJid()

?

**Comment by Flowdalic** [comment]
Thanks for you contribution. Please

- squash the commits into one and update this PR
- mention the issue key SMACK-619 somewhere in the body of the commit message

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/DefaultUserStatusListener.java (hunk 1 above)
Comment (meisterfuu) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/DefaultUserStatusListener.java]: Same as here:
MultiUserChat.destroy(String reason, String alternateJID)

Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/DefaultUserStatusListener.java]: Ahh, makes sense then. And I see it's javadoc'ed at the listener, so it's fine.

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java (hunk 2 above)
Comment (meisterfuu) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChat.java]: mucUser.getDestroy().getJid() returns a string. I can change that later if you want the parsing done in some other place.

**Comment by meisterfuu** [comment]
Ok, I update the PR later when I get home.
Commiting with the github editor was not the best thing i ever did.^^

**Comment by meisterfuu** [comment]
Squashed the commits

**Comment by Flowdalic** [comment]
I've just pushed https://github.com/igniterealtime/Smack/commit/516e39767994e8cf4a1b21caef6a35ca5bcc6d16

Could you rebase you commit on it, so that we can get rid of the jid parsing code?

**Comment by Flowdalic** [comment]
Also please change

    public abstract void roomDestroyed(Jid alternateJID, String reason);

to

    public abstract void roomDestroyed(MultiUserChat alternativeMuc, String reason);

**Comment by meisterfuu** [comment]
I will do it when I get home later.

**Comment by meisterfuu** [comment]
https://github.com/igniterealtime/Smack/pull/36

**Comment by Flowdalic** [comment]
FYI, no need to close a PR. You could simply update the existing one by force-pushing to its source branch.

**Comment by meisterfuu** [comment]
I messed with the master and wanted a clean clone from the smack master without a "Merge commit". I rarely do other things than Push/Pull with git, still learning :)

**Pull request closed without merging**